### PR TITLE
Companion PR for pytorch/pytorch#89326

### PR DESCRIPTION
**Pull request by nkaretnikov**

### CODE REVIEW

**Review by JackCaoG — APPROVED** [review]
Thanks! Please give me a heads up when you plan to merge upstream, I will merge this one right after.

**Comment by JackCaoG** [comment]
There was another pytorch/xla breaking pr merged to pytorch, please rebase both pytorch and xla pr before landing.

**Comment by nkaretnikov** [comment]
Created a new PR since I now have push access to the repo.

**Pull request closed without merging**
